docs: Improve example, API reference and intro (#24)
OpenDataServices/spreadsheet-forms: docs/examples.rst

========

A Guide Form spreadsheet
------------------------

.. |_| unicode:: 0xA0
   :trim:

> A guide form spreadsheet is the template to show us where to expect data. Any operation will need one of these to guide it.

Reviewer: ```suggestion
A guide form spreadsheet is a template that specifies the structure of a spreadsheet form. Most functions require a guide form.
```

Author: All functions require a guide form or spec

> These have special values in certain cells - this tells the library where and how to manipulate data.

Reviewer: ```suggestion
Guide forms use special values in certain cells to specify the structure of a spreadsheet form.
```

| SPREADSHEETFORM:DOWN:likes/toys:title  |  SPREADSHEETFORM:DOWN:likes/toys:squeak  |
+----------------------------------------+------------------------------------------+

Extracting data from a spreadsheet
----------------------------------

> Given the guide spreadsheet above and a spreadsheet with data in it, like:

Reviewer: ```suggestion
Given the guide form above and the following populated spreadsheet form:
```

Note the SINGLE keyword is turned into a field, but the DOWN row is turned into a list.

> People can add as many or as few items to the DOWN table as they want.

Reviewer: ```suggestion
Users can add as many items to the DOWN table as they want.
```

Author: I know Users is kinda industry standard term but it's also not a great term. I'd prefer to leave it. If we are going to clarify this at all, we should acknowledge that People/Users could refer to one of several groups of people and we should say which. Something like "The people filling in the spreadsheet can ..."

Reviewer: "The people filling in the spreadsheet can..." sounds good, just "People..." is too informal and isn't clear about *which* people.

> Putting data into a spreadsheet

Reviewer: ```suggestion
Populating a spreadsheet form
```

-------------------------------

The process can be run in reverse using the :doc:`put_data_in_form<api/put_data_in_form>` function.

> Given the JSON data above, the function will produce the spreadsheet above.

Reviewer: ```suggestion
Given the JSON data above, the function will produce the populated spreadsheet form above.
```
